Use upgrader_mobile.cpp as the reference for codegen unittest

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71930
* #71938

Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run @mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

Commit:
Use upgrader_mobile.cpp as the reference for codegen unittest

Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run @mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

tugsbayasgalan commented on test/mobile/test_upgrader_codegen.py:
```diff
@@ -11,15 +11,14 @@
 from torch.jit.generate_bytecode import generate_bytecode
 
-pytorch_test_dir = Path(__file__).resolve().parents[1]
+pytorch_caffe2_dir = Path(__file__).resolve().parents[2]
```
would this still work with OSS?

cccclai replied:
Yeah in OSS, it will point to the root folder like `/Users/chenlai/pytorch`.

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

tugsbayasgalan commented on torch/csrc/jit/mobile/upgrader_mobile.cpp:
```diff
@@ -270,3 +317,4 @@ const std::vector<ByteCodeFunctionWithOperator>& getUpgraderBytecodeList() {
 
```
where is clang-format-off?

cccclai replied:
`clang-format-off` is added in https://github.com/pytorch/pytorch/pull/71578 and it's merged. I just rebase

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]

Commit:
Update on "Use upgrader_mobile.cpp as the reference for codegen unittest"


Previously `fbcode/caffe2/test/mobile/test_upgrader_bytecode_table_example.cpp` was checked in as intermediate step to make sure upgrader codegen works properly, before upgrader codegen is actually being used.

this change use `buck run mode/opt //caffe2/torch/fb/mobile/upgrader_codegen:upgrader_codegen` to codegen `upgrader_mobile.cpp` and we no longer need to use the checkin file `test_upgrader_bytecode_table_example.cpp` for the codegen unit test.

Differential Revision: [D33746264](https://our.internmc.facebook.com/intern/diff/D33746264/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33746264/)!

[ghstack-poisoned]
